Yvand.EntraCP/Yvand.EntraClaimsProvider/EntraIDEntityProvider.cs

```diff
@@ -352,7 +362,7 @@ protected async Task<List<DirectoryObject>> QueryEntraIDTenantsAsync(OperationCo
             return allResults;
         }
 
-        protected virtual async Task<List<DirectoryObject>> QueryEntraIDTenantAsync(OperationContext currentContext, EntraIDTenant tenant)
+        protected virtual async Task<List<DirectoryObject>> QueryEntraIDTenantAsync(OperationContext currentContext, EntraIDTenant tenant, CachedEntraIDTenantData cachedTenantData)
```

Comment:
This method could really use some refactoring into smaller ones :)

Reply:
I guess you're right, but that would be for another day 😅

Reply:
Yup, definitely.